internal/service/streamaccountdetails/model.go

```diff
@@ -0,0 +1,29 @@
+package streamaccountdetails
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20250312003/admin"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func NewTFStreamAccountDetails(
+	ctx context.Context,
+	projectID string,
+	cloudProvider string,
+	region string,
+	accountDetails *admin.AccountDetails,
+) (*TFStreamAccountDetailsModel, diag.Diagnostics) {
+	return &TFStreamAccountDetailsModel{
+		ProjectId:           types.StringValue(projectID),
+		AwsAccountId:        types.StringPointerValue(accountDetails.AwsAccountId),
+		AzureSubscriptionId: types.StringPointerValue(accountDetails.AzureSubscriptionId),
+		CidrBlock:           types.StringPointerValue(accountDetails.CidrBlock),
+		CloudProvider:       types.StringValue(cloudProvider),
+		RegionName:          types.StringValue(region),
+		VirtualNetworkName:  types.StringPointerValue(accountDetails.VirtualNetworkName),
```

Comment:
q (not fully related to data source implementation): Do we know why `virtualNetworkName` is not showing in API docs? https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Streams/operation/getAccountDetails

Reply:
<img width="834" alt="Screenshot 2025-05-30 at 7 35 32 AM" src="https://github.com/user-attachments/assets/15e37a06-2421-4d55-8296-9eb1d289e5a8" />

It looks like a quirk of the docs where it won't show up unless the cloudProvider is selected in the response schema. It's a bit confusing though I imagine the intention was to show that the response fields depend on the cloud provider

Reply:
Oh I see, that makes sense, thanks for clarifying 👍